src/test.js

```diff
@@ -1,36 +1,31 @@
-export const test = (dataTypeCheck, algorithm, result, sortType="unknown", index) => {
+const chalk = require('chalk');
+
+const test = (dataTypeCheck, algorithm, dataset, index, sortType="unknown") => {
+    if (!dataTypeCheck(dataset)) {
+        console.log(chalk.yellow(`${JSON.stringify(dataset)} at #${index} is an invalid set of data for ${sortType}`));
+        return;
+    }
+
+    const result = dataset.sort((a,b) => a-b)
```

Comment:
can you please explain dataset.sort((a,b) => a-b) ? 
what is the idea here? I am not familiar with this

Reply:
[array].sort takes in a callback parameter that has two parameters, usually called 'a' and 'b'. They represent the two items compared at each time. The callback function itself can return -1, 0 or 1 behind the scenes. It is possible to write a long and complicated callback for your sort method as well, but what this one-line version does is that it returns -1 if the evaluation is negative, 0 if it is 0 and 1 if it is positive. I think that js loops through the array and swaps items as many times as necessary until the callback returns -1 for all compared pairs.

Without this, the sort method would compare the unicode of each number character, which is great for strings, but for numbers, it would not work. For example, it would say that [1, 120, 23] is a sorted array, since it is in correct alphabetical order.

the ((a,b) => a-b) callback given to the sort method means I want to sort numbers and I want to sort them in ascending order.